ext/standard/dns.c

```diff
@@ -415,13 +413,16 @@ PHP_FUNCTION(dns_check_record)
 #endif
 
 	RETVAL_TRUE;
-	i = php_dns_search(handle, hostname, C_IN, type, ans, sizeof(ans));
+	i = php_dns_search(handle, hostname, C_IN, type, answer.qb2, sizeof answer);
+	php_dns_free_handle(handle);
 
 	if (i < 0) {
 		RETVAL_FALSE;
 	}
-
-	php_dns_free_handle(handle);
+	hp = (HEADER *)&answer;
+	if (ntohs(hp->ancount) == 0) {
+		RETVAL_FALSE;
+	}
```

Comment:
I'd drop the RETVAL_TRUE above and write `RETURN_BOOL(ntohs(hp->ancount) != 0);` here. I think that would be clearer.

Reply:
Thank you a lot for review, updated commit!

btw Is it ok to keep PR as single commit?

Reply:
I mean it could be split on bug-fix, handle fix and clean-up...

Reply:
Also I removed `RETVAL_TRUE` before calling function